src/ClientMessage.ts

```diff
@@ -15,248 +15,231 @@
  */
 
 /* tslint:disable:no-bitwise */
-/*
- Client Message is the carrier framed data as defined below.
- Any request parameter, response or event data will be carried in the payload.
- 0                   1                   2                   3
- 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
- +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
- |R|                      Frame Length                           |
- +-------------+---------------+---------------------------------+
- |  Version    |B|E|  Flags    |               Type              |
- +-------------+---------------+---------------------------------+
- |                       CorrelationId                           |
- |                                                               |
- +---------------------------------------------------------------+
- |                        PartitionId                            |
- +-----------------------------+---------------------------------+
- |        Data Offset          |                                 |
- +-----------------------------+                                 |
- |                      Message Payload Data                    ...
- |                                                              ...
- */
-
 import {Buffer} from 'safe-buffer';
 import * as Long from 'long';
 import {BitsUtil} from './BitsUtil';
-import {Data} from './serialization/Data';
-import {HeapData} from './serialization/HeapData';
+import {ClientConnection} from './network/ClientConnection';
 
-class ClientMessage {
+export const MESSAGE_TYPE_OFFSET = 0;
```

Comment:
We had to expose `PARTITION_ID_OFFSET` and `RESPONSE_BACKUP_ACKS_OFFSET` since they are used in the initial frame size calculations in the codecs. I removed `export` from other offsets.